drgn/helpers/linux/waitq.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+Waitqueue
+--------------
+
+The ``drgn.helpers.linux.waitq`` module provides helpers for working with the
+wait-queue mechanism, implemented in :linux:`include/linux/wait.h`
+(``struct wait_queue_head`` and ``struct wait_queue_entry``).
+"""
+
+from typing import Iterator
+
+from drgn import Object, Program, Type
+from drgn.helpers.linux.list import list_empty, list_for_each_entry
+
+__all__ = (
+    "waitqueue_active",
+    "for_each_entry_in_waitqueue",
+    "for_each_task_in_waitqueue",
```

Comment:
I'd prefer the names `waitqueue_for_each_entry()` and `waitqueue_for_each_task()`; they'd be a little easier to discover via tab-completion.

Reply:
Agree. I have made these changes.